### PR TITLE
Clarified move semantics in "the details" section.

### PR DESCRIPTION
**Pull request by sandeep-datta**
See title and diff for more information.

### CODE REVIEW

**Comment by SDX2000** [comment]
@steveklabnik any idea why the CI builds are failing? The build seems to pass on my machine.

File: src/doc/book/ownership.md [hunk 1]
```diff
@@ -130,13 +142,41 @@ let v = vec![1, 2, 3];
...
+Which means that there would be two pointers to the contents of the vector
+both pointing to the same memory allocation on the heap. It would violate
+Rust’s safety guarantees by introducing a data race if one could access both
+`v` and `v2` at the same time. 
+
+For example if we truncated the vector to just two elements through `v2`:
+
+```rust
+v2.truncate(2);
+```
```
Comment (steveklabnik) [inline src/doc/book/ownership.md]: This is causing the travis failure:

```

---- Move_semantics_5 stdout ----
	<anon>:2:5: 2:7 error: unresolved name `v2` [E0425]
<anon>:2     v2.truncate(2);
             ^~
```

You should change this to
```text
    ```rust
    # let v = vec![1, 2, 3];
    # let v2 = v;
    v2.truncate(2);
    ```
```

**Comment by steveklabnik** [comment]
It's gotta be `mut` for `truncate()` to work

**Comment by sandeep-datta** [comment]
@steveklabnik 

Okay so I changed the code to...
```
fn main() {
    let v = vec![1, 2, 3];
    let mut v2 = v;
    v2.truncate(2);
    println!("{:?}", v2);
}
```

And it worked.

But now I wonder if we have to switch to a mutable reference to actually inflict any damage then why does rust object to using older immutable references? Is this simply to make book keeping easier for the compiler?

**Comment by steveklabnik** [comment]
I'm not sure what you mean, but I _think_ I do, so let me try:

Each triple-backtick block is its own little universe. There's no shared context between them. So there is no 'older immutable binding' lying around.

Does that make sense? maybe I'm still confused :smile: 

regardless, @bors: r+ rollup

**Comment by sandeep-datta** [comment]
> Each triple-backtick block is its own little universe.

I guessed as much from your suggested fix. 

But this isn't what I was asking. I wanted to know why does the compiler object to reusing a binding once it has been "moved"? Considering the fact that a "move" really is only a bitwise copy and if everything remains immutable then what problem are we trying to avoid by disallowing post-move access to a binding?

**Comment by pnkfelix** [comment]
@sandeep-datta :

after moving the vector into `v2`, `v2` now has ownership.

The vector thus could later be moved elsewhere (perhaps into a binding that is declared as `mut`, which would re enable mutation) and perhaps dropped entirely. 

Example:

```rust
fn main() {
    let v = vec![1, 2, 3];
    let v2 = v;
    {   let v3 = v2; drop(v3); } // explicit drop for clarity, 
        // but even without it v3 would be dropped at end of this inner scope
    println!("{:?}", v);
}
```

**Comment by KalitaAlexey** [comment]
@sandeep-datta Consider `UnsafeCell` instead of `Vec`. It is externally immutable, but internally mutable. It may lead to errors.

**Comment by sandeep-datta** [comment]
@pnkfelix @KalitaAlexey thanks! I will update the documentation to demonstrate this problem using UnsafeCell.

**Pull request merged**